sn_interface/src/messaging/serialisation/wire_msg.rs

```diff
@@ -278,11 +315,15 @@ impl WireMsg {
 
 #[cfg(feature = "traceroute")]
 impl WireMsg {
-    pub fn add_trace(&mut self, traceroute: &mut Vec<Entity>) {
-        self.header.msg_envelope.traceroute.append(traceroute)
+    pub fn add_trace(&mut self, traceroute: &mut Traceroute) {
+        self.header
+            .msg_envelope
+            .traceroute
+            .0
+            .append(&mut traceroute.0)
     }
 
-    pub fn show_trace(&self) -> Vec<Entity> {
+    pub fn show_trace(&self) -> Traceroute {
```

Comment:
`show_trace()` implies this function would print or somehow display a traceroute, but it's just accessing the trace
```suggestion
    pub fn trace(&self) -> Traceroute {
```